components/tables/TableCell.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState, useEffect, ChangeEvent } from "react";
+import { CellContext, Column } from "@tanstack/react-table";
+import { User } from "@/types/types";
+import { ColumnType } from "./AdminTable";
+
+interface TableCellProps<TData> extends CellContext<TData, unknown> {
+  column: Column<TData, unknown>;
+}
+
+const TableCell = <TData,>({
+  getValue,
+  row,
+  column,
+  table,
+}: TableCellProps<TData>) => {
+  const initialValue = getValue() as any;
+  const columnMeta = column.columnDef.meta;
+  const tableMeta = table.options.meta as any;
+  const [value, setValue] = useState<any>(initialValue);
+
+  useEffect(() => {
+    if (column.id === "password") {
+      setValue(""); // so it does not change password
+    } else {
+      setValue(initialValue);
+    }
+  }, [initialValue]);
+
+  const handleEdit = (e: ChangeEvent<any>) => {
+    const newValue = e.target.value;
+    setValue(newValue);
+
+    if (column.id === "hasPaid") {
+      if (
+        newValue == "True" &&
+        tableMeta.editFormData?.paymentMethod == "" &&
+        tableMeta.editFormData.verifier == "" &&
+        tableMeta.editFormData.paymentLocation == ""
+      ) {
+        tableMeta.setEditFormData((prevData: User | null) =>
+          prevData
+            ? {
+                ...prevData,
+                ["hasPaid"]: newValue,
+                ["verifier"]: name,
+              }
+            : null,
+        );
+      } else if (newValue == "True") {
+        tableMeta.setEditFormData((prevData: User | null) =>
+          prevData ? { ...prevData, ["hasPaid"]: newValue } : null,
+        );
+      } else {
+        tableMeta.setEditFormData((prevData: User | null) =>
+          prevData
+            ? {
+                ...prevData,
+                ["hasPaid"]: newValue,
+                ["paymentMethod"]: "",
+                ["verifier"]: "",
+                ["paymentLocation"]: "",
+              }
+            : null,
+        );
+      }
+    } else {
+      tableMeta.setEditFormData((prevData: User | null) =>
+        prevData ? { ...prevData, [column.id]: newValue } : null,
+      );
+      console.log("Edit Form Data:", tableMeta.editFormData);
+    }
+  };
+
+  const user = row.original as User;
```

Comment:
This variable is not used, maybe we could remove it ? I didn't remove it myself incase it relates to the next comment.

Reply:
yep unused - removed